posthog/temporal/batch_exports/squash_person_overrides.py

```diff
@@ -161,6 +132,272 @@ class SerializablePersonOverrideToDelete(NamedTuple):
     oldest_event_at: str
 
 
+class PersonOverrideTuple(NamedTuple):
+    old_person_id: UUID
+    override_person_id: UUID
+
+
+class PostgresPersonOverridesManager:
+    def __init__(self, connection):
+        self.connection = connection
+
+    def fetchall(self, team_id: int) -> Sequence[PersonOverrideTuple]:
+        with self.connection.cursor() as cursor:
+            cursor.execute(
+                """
+                SELECT
+                    old_person.uuid,
+                    override_person.uuid
+                FROM posthog_personoverride override
+                LEFT OUTER JOIN
+                    posthog_personoverridemapping old_person
+                    ON override.team_id = old_person.team_id AND override.old_person_id = old_person.id
+                LEFT OUTER JOIN
+                    posthog_personoverridemapping override_person
+                    ON override.team_id = override_person.team_id AND override.override_person_id = override_person.id
+                WHERE override.team_id = %(team_id)s
+                """,
+                {"team_id": team_id},
+            )
+            return [PersonOverrideTuple(*row) for row in cursor.fetchall()]
+
+    def insert(self, team_id: int, override: PersonOverrideTuple) -> None:
+        with self.connection.cursor() as cursor:
+            mapping_ids = []
+            for person_uuid in (override.override_person_id, override.old_person_id):
+                cursor.execute(
+                    """
+                    INSERT INTO posthog_personoverridemapping(
+                        team_id,
+                        uuid
+                    )
+                    VALUES (
+                        %(team_id)s,
+                        %(uuid)s
+                    )
+                    ON CONFLICT("team_id", "uuid") DO NOTHING
+                    RETURNING id
+                    """,
+                    {"team_id": team_id, "uuid": person_uuid},
+                )
+                mapping_ids.append(cursor.fetchone())
+
+            cursor.execute(
+                """
+                INSERT INTO posthog_personoverride(
+                    team_id,
+                    old_person_id,
+                    override_person_id,
+                    oldest_event,
+                    version
+                )
+                VALUES (
+                    %(team_id)s,
+                    %(old_person_id)s,
+                    %(override_person_id)s,
+                    NOW(),
```

Comment:
Because this is only for tests, right? Otherwise we'd use something from one of the users?

Reply:
Yep!